include/complex_quda.h

```diff
@@ -511,6 +519,14 @@ struct complex <float> : public cuFloatComplex
       return *this;
     }
 
+  __host__ __device__
```

Comment:
There are a couple of functions like
```
inline complex<float>& operator-=(const complex<float> z)
```

which could be

```
template<class T>
inline complex<T>& operator-=(const complex<T> z)
```

Same for things like operations between `complex<T>` and `T`, e.g.
```
inline complex<float>& operator*=(const float z)
```